### PR TITLE
Add support for building the archive from separate content branches.

### PR DESCRIPTION
**Pull request by nuclearsandwich**
This PR finally incorporates the Answers site contents via dedicated orphan content branches (already pushed).

I ended up adding an Earthly build along the way as I was building towards testing the redirect handlers in the generated Apache / nginx configs but that's still incomplete so I'm saving it for another PR. I've updated the readme to provide the Earthly build instructions first but left the local development instructions intact as an alternative.

I also removed `content/question/1.md`, the test question that we'd be using for the archive format test, as well as the saved files in `data` and the questions data source code which is no longer used.

### CODE REVIEW

File: nanoc.yaml [hunk 1]
```diff
@@ -5,8 +5,6 @@
 # will be considered as binary.
 text_extensions: ['html', 'md', 'inc', 'conf']
 
-site_title: Gazebo
-
```
Comment (Crola1702) [inline nanoc.yaml:9]: Should we make this a placeholder instead so `site_title` is not omitted?

File: lib/question_data.rb
Comment (Crola1702) [inline lib/question_data.rb:1]: :cry:

File: build.earth
Comment (Crola1702) [inline build.earth:1]: Nice!

File: checkout_content.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	*)
+		>&2 echo "No content branch for $1. Specify `ros` or `gazebo`."
+
+		exit 1
+esac
+
+git checkout .
+git clean -fx content/question/ nanoc.yaml
+git restore --source $BRANCH content/question/
+git show $BRANCH:content_config.yml >> nanoc.yaml
```
Comment (Crola1702) [inline checkout_content.sh:33]: Ahh, here is the `site_title` variable defined. Clever!

Nevermind my comment above

**Review by Crola1702 — APPROVED** [review]

File: lib/question_data.rb
Comment (nuclearsandwich) [reply lib/question_data.rb:1]: We learned a lot! And custom data sources are still going to support the Infrastructure web site as well as others in the future. 

I had a cursed idea to have nanoc generate terraform configs from rosdistro data sources 😁

**Pull request merged**